scene/animation/animation_mixer.cpp

```diff
@@ -2138,3 +2138,82 @@ AnimationMixer::AnimationMixer() {
 
 AnimationMixer::~AnimationMixer() {
 }
+
+void AnimatedValuesBackup::set_data(const HashMap<NodePath, AnimationMixer::TrackCache *> p_data) {
+	clear_data();
+
+	for (const KeyValue<NodePath, AnimationMixer::TrackCache *> &E : p_data) {
+		data.insert(E.key, get_cache_copy(E.value));
+	}
+}
+
+HashMap<NodePath, AnimationMixer::TrackCache *> AnimatedValuesBackup::get_data() const {
+	HashMap<NodePath, AnimationMixer::TrackCache *> ret;
+	for (const KeyValue<NodePath, AnimationMixer::TrackCache *> &E : data) {
+		ret.insert(E.key, get_cache_copy(E.value));
+	}
+	return ret;
+}
+
+void AnimatedValuesBackup::clear_data() {
+	for (KeyValue<NodePath, AnimationMixer::TrackCache *> &K : data) {
+		memdelete(K.value);
+	}
+	data.clear();
+}
+
+AnimationMixer::TrackCache *AnimatedValuesBackup::get_cache_copy(AnimationMixer::TrackCache *p_cache) const {
+	switch (p_cache->type) {
+		case Animation::TYPE_VALUE: {
+			AnimationMixer::TrackCacheValue *src = static_cast<AnimationMixer::TrackCacheValue *>(p_cache);
+			AnimationMixer::TrackCacheValue *tc = memnew(AnimationMixer::TrackCacheValue);
+			memcpy((void *)tc, src, sizeof(AnimationMixer::TrackCacheValue));
```

Comment:
I'm not sure it'll be safe to ignore this warning, I don't think it's overly careful saying not to memcpy virtual classes

Reply:
How it should be handled otherwise?

Reply:
With a copy constructor or copy assignment, as per the error suggestion I'd say

Reply:
From what I understand, this is only a problem if we ever add a new cache type that inherits a *subtype* AND the new type will use an existing enum value. It's unlikely and that would be a bug in itself, so it's not really a concern IMO.

Reply:
As long as the copying of the vtable is safe

Reply:
The HashMap is not copied, it's recreated. Only the track cache pointers are copied.

Reply:
How do you mean? The audio track cache has a hash map in it, how is it not copied?

Reply:
If that is a problem, it may be safe not to copy for the audio, method, and animation tracks although I think the more problem is that the Reset track is allowed for them in the first place.

Reply:
It's be safe to copy just as long as you do a copy and not just a raw copy of the memory, copying the memory stored in a hash map is unsafe as it uses raw data

Reply:
I haven't tested it as I'm not able to right now but my assumption here is that the members of these classes are copied wholesale byte by byte, but the problem with that is that at least `HashMap` is going to be unsafe *if* it is copied that way, as it handles memory by pointers, in the "best" case the resulting copied map would share data with the original, which at minimum causes problems with manipulation of one, might cause crashes due to invalid pointer access in the worst case

I also am not sure it's safe to memcpy a `Ref` as it should do refcounting and I suspect you'd have a double-reference in two elements, so no additional referencing would happen, causing extra freeing

Reply:
I agree so much with @AThousandShips that I've made #85302.